config-builder/src/main/java/com/ca/apim/gateway/cagatewayconfig/bundle/builder/AnnotatedBundle.java

```diff
@@ -58,12 +60,33 @@ public void addDependentBundle(DependentBundle dependentBundle){
         return dependentBundles;
     }
 
-    public String getUniquePrefix() {
-        return projectInfo.getName() + uniqueNameSeparator + PathUtils.extractName(annotatedEntity.getEntityName()) + uniqueNameSeparator;
-    }
+    /**
+     * Applies unique name space and version to the given entity name
+     * ::groupName.bundleName::entityName::majorVersion.minorVersion
+     *
+     * @param entityName String
+     * @return String
+     */
+    public String applyUniqueName(final String entityName) {
+        StringBuilder uniqueName = new StringBuilder(uniqueNameSeparator);
+        if (StringUtils.isNotBlank(projectInfo.getGroupName())) {
+            uniqueName.append(projectInfo.getGroupName());
+            uniqueName.append(".");
+        }
+        uniqueName.append(getAnnotatedBundleName());
```

Comment:
I assume shared or [reusable] annotation will be taken care of later. The entity that is shared across multiple bundles shouldn't include bundle names in it.

Reply:
Yes, we will handle shared entities in other story